packages/node_modules/pouchdb-adapter-indexeddb/src/index.js

```diff
@@ -153,6 +176,130 @@ function IndexeddbPouch(dbOpts, callback) {
+          if (seen.has(rev)) {
+            continue;
+          }
+          seen.set(rev);
+
+          if (!(rev in metadata.revs)) {
+            result.push({ missing: leaf });
+          } else {
+            const doc = metadata.revs[rev].data;
+            doc._id = metadata.id;
+            doc._rev = rev;
+
+            if (isDeleted(metadata, rev)) {
+              doc._deleted = true;
+            }
+
+            if (opts.revs) {
+              processRevsRevsInfo(id, doc, metadata.rev_tree, opts);
+            }
+
+            processing.push(new Promise((resolve, reject) => {
+              const cb = (err, res) => err ? reject(err) : resolve(res);
+              processAttachments(api, metadata, doc, opts, NO_CTX_REQUIRED, cb);
+            }));
+
+            result.push({ ok: doc });
+          }
+        }
+        try {
+          await Promise.all(processing);
+          cb(null, result);
+        } catch (err) {
+          cb(err);
+        }
+      };
+
+      api._openTransactionSafely([DOC_STORE], 'readonly', function (err, txn) {
+        txn.onabort = function () {
+          cb(createError(UNKNOWN_ERROR, 'transaction was aborted'));
+        };
+        txn.ontimeout = idbError(cb);
+        var req = txn.objectStore(DOC_STORE).get(id);
+        req.onsuccess = function (e) {
+          metadata = e.target.result;
+          if (opts.open_revs === "all") {
+            leaves = collectLeaves(metadata.rev_tree).map(function (leaf) {
+              return leaf.rev;
+            });
+            finishOpenRevs();
+          } else {
+            if (Array.isArray(opts.open_revs)) {
+              leaves = opts.open_revs;
+              for (var i = 0; i < leaves.length; i++) {
+                var l = leaves[i];
+                // looks like it's the only thing couchdb checks
+                // TODO replace with !isValidRev(l);
+                if (!(typeof (l) === "string" && /^\d+-/.test(l))) {
+                  return cb(createError(INVALID_REV));
+                }
+              }
+              finishOpenRevs();
+            } else {
+              return cb(createError(UNKNOWN_ERROR, 'function_clause'));
+            }
+          }
+        };
+      });
+      return; // open_revs does not like other options
+    }
+
+    return api._get(id, opts, (err, result) => {
+      if (err) {
+        err.docId = id;
+        return cb(err);
+      }
+
+      var doc = result.doc;
+      var metadata = result.metadata;
+      var ctx = result.ctx;
+
+      if (opts.conflicts) {
+        var conflicts = collectConflicts(metadata);
+        if (conflicts.length) {
+          doc._conflicts = conflicts;
+        }
+      }
+
+      if (isDeleted(metadata, doc._rev)) {
+        doc._deleted = true;
+      }
+
+      processRevsRevsInfo(id, doc, metadata.rev_tree, opts);
+
+      processAttachments(api, metadata, doc, opts, ctx, cb);
+    });
+  }).bind(api);
```

Comment:
This doesn't look like code you would write... is it copied from somewhere? If so, can it be reused?

Reply:
IIRC it mostly came from `adapter.js` in `pouchd-core`.  You're right that there's probably more code that could be shared